inc/saitypes.h

```diff
@@ -944,6 +944,88 @@ typedef struct _sai_port_eye_values_list_t
     sai_port_lane_eye_values_t *list;
 } sai_port_eye_values_list_t;
 
+/**
+ * @brief System port configuration attributes
+ */
+typedef struct _sai_system_port_config_t
+{
+    uint32_t port_id;
+    uint32_t attached_switch_id;
+    uint32_t attached_core_index;
+    uint32_t attached_core_port_index;
+    uint32_t speed;
+    uint32_t num_voq;
+} sai_system_port_config_t;
+
+/**
+ * @brief System port configuration list
+ */
+typedef struct _sai_system_port_config_list_t
+{
+    /** Number of entries in the list */
+    uint32_t count;
+
+    /** System port configuration list */
+    sai_system_port_config_t *list;
+} sai_system_port_config_list_t;
+
+/**
+ * @brief Fabric port reachability
+ */
+typedef struct _sai_fabric_port_reachability_t
+{
+    /** Remote switch ID (SAI_SWITCH_TYPE_NPU) */
+    uint32_t switch_id;
+
+    /** Remote switch ID is reachable through the fabric port */
+    bool reachable;
+} sai_fabric_port_reachability_t;
+
+/**
+ * @brief Port error status
+ */
+typedef enum _sai_port_err_status_t
```

Comment:
If these errors are specific to fabric ports, should we call that out in the name  - sai_fabric_port_err_status_t ?

Reply:
In the VoQ context, these apply to fabric ports but the errors are generic enough for front panel ports as well.   For example:  CRC, FEC, remote fault can apply to front panel ports as well.